src/col/vct/col/resolve/lang/Spec.scala

```diff
@@ -15,7 +15,7 @@ case object Spec {
     case RefInstanceMethod(decl) => decl.contract
     case RefCLocalDeclaration(decls, _) => decls.decl.contract
     case RefCGlobalDeclaration(decls, _) => decls.decl.contract
-//    case RefCFunctionDefinition(decl) =>
+    case RefCFunctionDefinition(decl) => decl.contract
```

Comment:
I think this is commented out because the contract of a function definition may also be the contract of the pre-declaration, if it is empty at the definition.

Reply:
Okay, so it should be something like `decl.contract ? decl.contract : pre_decl.contract`? What is the pre-declaration / where does it come from? nothing in `CFunctionDefinition` immediately looks fitting...

Reply:
haha hence why it was not immediately solved :) I don't think it's directly available here. I'll have a look on what to do

Reply:
Actually there seems to be no logic at all connecting forward declarations and definitions, so I'm ok leaving it like this for now.